WcaOnRails/app/webpacker/components/Panel/PanelTemplate.jsx

```diff
@@ -3,21 +3,39 @@ import {
   Dropdown,
   Grid, Header, Icon, Menu, Segment,
 } from 'semantic-ui-react';
+import useHash from '../../lib/hooks/useHash';
 
 export default function PanelTemplate({ heading, sections }) {
-  const [selectedMenu, setSelectedMenu] = React.useState(0);
-  const SelectedComponent = React.useMemo(() => sections.find(
-    (_, index) => index === selectedMenu,
-  ).component, [sections, selectedMenu]);
+  const [hash, setHash] = useHash();
 
-  function menuClickHandler(index) {
-    const section = sections[index];
+  const selectedMenu = React.useMemo(() => (hash ? sections.findIndex(
+    (section) => section.id === hash,
+  ) : 0), [hash, sections]);
+
+  if (selectedMenu === -1) {
+    setHash(sections[0].id);
+  }
+
+  const SelectedComponent = React.useMemo(() => {
+    const selectedSectionIndex = sections.findIndex((section) => section.id === hash);
+    const selectedSection = sections[selectedSectionIndex] || sections[0];
+    if (selectedSectionIndex === -1) {
+      setHash(selectedSection.id);
+    }
+    if (selectedSection.component) {
+      return selectedSection.component;
+    }
+    window.location.href = selectedSection.link;
```

Comment:
Why are you overriding the browser's location href? Shouldn't a simple `setHash` be enough?
(I thought that's the whole reason why we introduced the `useHash` hook in the first place, to avoid manually setting location hrefs?!)